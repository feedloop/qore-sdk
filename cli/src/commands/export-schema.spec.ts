import { setupRecorder } from "nock-record";
import fs from "fs";
import path from "path";
import ExportSchema from "./export-schema";

const record = setupRecorder();

describe("export-schema", () => {
  it("should be able to export schema", async () => {
    process.env.REACT_APP_API_URL = 'https://p-qore-dot-pti-feedloop.et.r.appspot.com';
    const { completeRecording } = await record("export-schema");
    await ExportSchema.run([
      "--project",
      "I0D3NimZQ9GKEDP",
      "--org",
      "lIdfC42DJCN2XzQ",
      "--token",
      "77f2ff71-8864-404d-8596-127d78a4c1bd",
    ]);
    completeRecording();
    const filename = path.resolve(process.cwd() + "/qore-schema.json");
    expect(
      fs.readFileSync(filename, {
        encoding: "utf8",
      })
    ).toMatchSnapshot();
    fs.unlinkSync(filename);
  });
});
