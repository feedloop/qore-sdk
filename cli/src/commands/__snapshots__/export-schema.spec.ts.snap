// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`export-schema should be able to export schema 1`] = `
"{
  \\"version\\": \\"v1\\",
  \\"tables\\": [
    {
      \\"id\\": \\"member\\",
      \\"name\\": \\"member\\",
      \\"type\\": \\"auth\\",
      \\"fields\\": [
        {
          \\"id\\": \\"role\\",
          \\"name\\": \\"role\\",
          \\"linked\\": false,
          \\"type\\": \\"role\\",
          \\"deletionProtection\\": true,
          \\"createdAt\\": \\"2020-12-08T07:31:04.801Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"email\\",
          \\"name\\": \\"email\\",
          \\"linked\\": false,
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": true,
          \\"createdAt\\": \\"2020-12-08T07:31:04.801Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"toDo\\",
          \\"name\\": \\"To Do\\",
          \\"linked\\": false,
          \\"type\\": \\"relation\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:02:47.440Z\\",
          \\"table\\": \\"toDo\\",
          \\"multiple\\": true
        },
        {
          \\"id\\": \\"totalTask\\",
          \\"name\\": \\"Total task\\",
          \\"linked\\": false,
          \\"type\\": \\"rollup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:37:51.822Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"totalPoint\\",
          \\"name\\": \\"Total point\\",
          \\"linked\\": false,
          \\"type\\": \\"rollup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:38:17.873Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"totalHardTask\\",
          \\"name\\": \\"Total hard task\\",
          \\"linked\\": false,
          \\"type\\": \\"rollup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:42:13.651Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"createTask\\",
          \\"name\\": \\"Create task\\",
          \\"linked\\": false,
          \\"type\\": \\"action\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:49:35.850Z\\",
          \\"refRows\\": null
        }
      ],
      \\"forms\\": []
    },
    {
      \\"id\\": \\"toDo\\",
      \\"name\\": \\"To Do\\",
      \\"type\\": null,
      \\"fields\\": [
        {
          \\"id\\": \\"task\\",
          \\"name\\": \\"Task\\",
          \\"linked\\": false,
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:46.250Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"description\\",
          \\"name\\": \\"description\\",
          \\"linked\\": false,
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:47.255Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"done\\",
          \\"name\\": \\"Done\\",
          \\"linked\\": false,
          \\"type\\": \\"boolean\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:50:42.534Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"points\\",
          \\"name\\": \\"Points\\",
          \\"linked\\": false,
          \\"type\\": \\"number\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:51:15.322Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"deadline\\",
          \\"name\\": \\"Deadline\\",
          \\"linked\\": false,
          \\"type\\": \\"date\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:52:25.580Z\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"difficulty\\",
          \\"name\\": \\"Difficulty\\",
          \\"linked\\": false,
          \\"type\\": \\"select\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:00:20.824Z\\",
          \\"select\\": [\\"Easy\\", \\"Medium\\", \\"Hard\\"],
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"person\\",
          \\"name\\": \\"Person\\",
          \\"linked\\": false,
          \\"type\\": \\"relation\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:02:47.423Z\\",
          \\"table\\": \\"member\\",
          \\"multiple\\": true
        },
        {
          \\"id\\": \\"role\\",
          \\"name\\": \\"Role\\",
          \\"linked\\": false,
          \\"type\\": \\"lookup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:06:05.437Z\\",
          \\"returnType\\": \\"table\\",
          \\"multiple\\": false,
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"attachment\\",
          \\"name\\": \\"Attachment\\",
          \\"linked\\": false,
          \\"type\\": \\"file\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:06:33.320Z\\",
          \\"fileType\\": \\"image\\",
          \\"refRows\\": null
        },
        {
          \\"id\\": \\"timeAllocation\\",
          \\"name\\": \\"Time Allocation\\",
          \\"linked\\": false,
          \\"type\\": \\"formula\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:18:42.734Z\\",
          \\"returnType\\": \\"text\\",
          \\"refRows\\": null
        }
      ],
      \\"forms\\": [
        {
          \\"id\\": \\"newForm3MuYd8OVj6FdlgEnmUv\\",
          \\"name\\": \\"new form 3muYd8OVj-6fdlg-EnmUv\\",
          \\"tableId\\": \\"toDo\\",
          \\"_config\\": {
            \\"organizationId\\": \\"lIdfC42DJCN2XzQ\\",
            \\"projectId\\": \\"I0D3NimZQ9GKEDP\\",
            \\"token\\": \\"Bearer 77f2ff71-8864-404d-8596-127d78a4c1bd\\"
          },
          \\"fields\\": [
            { \\"id\\": \\"task\\", \\"defaultValue\\": null },
            { \\"id\\": \\"description\\", \\"defaultValue\\": null },
            { \\"id\\": \\"done\\", \\"defaultValue\\": null },
            { \\"id\\": \\"points\\", \\"defaultValue\\": null },
            { \\"id\\": \\"deadline\\", \\"defaultValue\\": null }
          ]
        }
      ]
    }
  ],
  \\"views\\": [
    {
      \\"id\\": \\"memberDefaultView\\",
      \\"name\\": \\"member Default View\\",
      \\"sorts\\": [],
      \\"parameters\\": [],
      \\"tableId\\": \\"member\\",
      \\"vields\\": [
        {
          \\"id\\": \\"email\\",
          \\"name\\": \\"email\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": true,
          \\"createdAt\\": \\"2020-12-08T07:31:04.801Z\\"
        },
        {
          \\"id\\": \\"role\\",
          \\"name\\": \\"role\\",
          \\"type\\": \\"role\\",
          \\"deletionProtection\\": true,
          \\"createdAt\\": \\"2020-12-08T07:31:04.801Z\\"
        },
        {
          \\"id\\": \\"createTask\\",
          \\"name\\": \\"Create task\\",
          \\"type\\": \\"action\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:49:35.850Z\\"
        }
      ]
    },
    {
      \\"id\\": \\"toDoDefaultView\\",
      \\"name\\": \\"Tasks\\",
      \\"sorts\\": [{ \\"order\\": \\"desc\\", \\"by\\": \\"points\\" }],
      \\"parameters\\": [],
      \\"tableId\\": \\"toDo\\",
      \\"vields\\": [
        {
          \\"id\\": \\"task\\",
          \\"name\\": \\"Task\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:46.250Z\\"
        },
        {
          \\"id\\": \\"done\\",
          \\"name\\": \\"Done\\",
          \\"type\\": \\"boolean\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:50:42.534Z\\"
        },
        {
          \\"id\\": \\"points\\",
          \\"name\\": \\"Points\\",
          \\"type\\": \\"number\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:51:15.322Z\\"
        },
        {
          \\"id\\": \\"deadline\\",
          \\"name\\": \\"Deadline\\",
          \\"type\\": \\"date\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:52:25.580Z\\"
        },
        {
          \\"id\\": \\"difficulty\\",
          \\"name\\": \\"Difficulty\\",
          \\"type\\": \\"select\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:00:20.824Z\\",
          \\"select\\": [\\"Easy\\", \\"Medium\\", \\"Hard\\"]
        },
        {
          \\"id\\": \\"person\\",
          \\"name\\": \\"Person\\",
          \\"type\\": \\"relation\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:02:47.423Z\\",
          \\"table\\": \\"member\\",
          \\"multiple\\": true
        },
        {
          \\"id\\": \\"role\\",
          \\"name\\": \\"Role\\",
          \\"type\\": \\"lookup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:06:05.437Z\\",
          \\"returnType\\": \\"table\\",
          \\"multiple\\": false
        },
        {
          \\"id\\": \\"attachment\\",
          \\"name\\": \\"Attachment\\",
          \\"type\\": \\"file\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:06:33.320Z\\",
          \\"fileType\\": \\"image\\"
        },
        {
          \\"id\\": \\"timeAllocation\\",
          \\"name\\": \\"Time Allocation\\",
          \\"type\\": \\"formula\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:18:42.734Z\\",
          \\"returnType\\": \\"text\\"
        }
      ]
    },
    {
      \\"id\\": \\"test\\",
      \\"name\\": \\"Priorities\\",
      \\"sorts\\": [],
      \\"parameters\\": [],
      \\"tableId\\": \\"toDo\\",
      \\"vields\\": [
        {
          \\"id\\": \\"task\\",
          \\"name\\": \\"Task\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:46.250Z\\"
        },
        {
          \\"id\\": \\"description\\",
          \\"name\\": \\"description\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:47.255Z\\"
        }
      ]
    },
    {
      \\"id\\": \\"toDo\\",
      \\"filter\\": \\"done = 'FALSE'\\",
      \\"name\\": \\"To Do\\",
      \\"sorts\\": [],
      \\"parameters\\": [],
      \\"tableId\\": \\"toDo\\",
      \\"vields\\": [
        {
          \\"id\\": \\"task\\",
          \\"name\\": \\"Task\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:46.250Z\\"
        },
        {
          \\"id\\": \\"description\\",
          \\"name\\": \\"description\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:47.255Z\\"
        },
        {
          \\"id\\": \\"done\\",
          \\"name\\": \\"Done\\",
          \\"type\\": \\"boolean\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:50:42.534Z\\"
        },
        {
          \\"id\\": \\"points\\",
          \\"name\\": \\"Points\\",
          \\"type\\": \\"number\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:51:15.322Z\\"
        },
        {
          \\"id\\": \\"deadline\\",
          \\"name\\": \\"Deadline\\",
          \\"type\\": \\"date\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:52:25.580Z\\"
        }
      ]
    },
    {
      \\"id\\": \\"done\\",
      \\"filter\\": \\"done = 'TRUE'\\",
      \\"name\\": \\"Done\\",
      \\"sorts\\": [],
      \\"parameters\\": [],
      \\"tableId\\": \\"toDo\\",
      \\"vields\\": [
        {
          \\"id\\": \\"task\\",
          \\"name\\": \\"Task\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:46.250Z\\"
        },
        {
          \\"id\\": \\"description\\",
          \\"name\\": \\"description\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:47.255Z\\"
        },
        {
          \\"id\\": \\"done\\",
          \\"name\\": \\"Done\\",
          \\"type\\": \\"boolean\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:50:42.534Z\\"
        },
        {
          \\"id\\": \\"points\\",
          \\"name\\": \\"Points\\",
          \\"type\\": \\"number\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:51:15.322Z\\"
        },
        {
          \\"id\\": \\"deadline\\",
          \\"name\\": \\"Deadline\\",
          \\"type\\": \\"date\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:52:25.580Z\\"
        }
      ]
    },
    {
      \\"id\\": \\"reisTask\\",
      \\"filter\\": \\"timeAllocation = $param.timeAllocation\\",
      \\"name\\": \\"Task by timeAllocation\\",
      \\"sorts\\": [],
      \\"parameters\\": [
        { \\"slug\\": \\"timeAllocation\\", \\"type\\": \\"text\\", \\"required\\": false }
      ],
      \\"tableId\\": \\"toDo\\",
      \\"vields\\": [
        {
          \\"id\\": \\"task\\",
          \\"name\\": \\"Task\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:46.250Z\\"
        },
        {
          \\"id\\": \\"description\\",
          \\"name\\": \\"description\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:44:47.255Z\\"
        },
        {
          \\"id\\": \\"done\\",
          \\"name\\": \\"Done\\",
          \\"type\\": \\"boolean\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:50:42.534Z\\"
        },
        {
          \\"id\\": \\"points\\",
          \\"name\\": \\"Points\\",
          \\"type\\": \\"number\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:51:15.322Z\\"
        },
        {
          \\"id\\": \\"deadline\\",
          \\"name\\": \\"Deadline\\",
          \\"type\\": \\"date\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T07:52:25.580Z\\"
        },
        {
          \\"id\\": \\"difficulty\\",
          \\"name\\": \\"Difficulty\\",
          \\"type\\": \\"select\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:00:20.824Z\\",
          \\"select\\": [\\"Easy\\", \\"Medium\\", \\"Hard\\"]
        },
        {
          \\"id\\": \\"person\\",
          \\"name\\": \\"Person\\",
          \\"type\\": \\"relation\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:02:47.423Z\\",
          \\"table\\": \\"member\\",
          \\"multiple\\": true
        },
        {
          \\"id\\": \\"role\\",
          \\"name\\": \\"Role\\",
          \\"type\\": \\"lookup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:06:05.437Z\\",
          \\"returnType\\": \\"table\\",
          \\"multiple\\": false
        },
        {
          \\"id\\": \\"attachment\\",
          \\"name\\": \\"Attachment\\",
          \\"type\\": \\"file\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:06:33.320Z\\",
          \\"fileType\\": \\"image\\"
        },
        {
          \\"id\\": \\"timeAllocation\\",
          \\"name\\": \\"Time Allocation\\",
          \\"type\\": \\"formula\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-08T08:18:42.734Z\\",
          \\"returnType\\": \\"text\\"
        }
      ]
    },
    {
      \\"id\\": \\"memberMetric\\",
      \\"filter\\": \\"email = $param.email\\",
      \\"name\\": \\"Member Metric\\",
      \\"sorts\\": [],
      \\"parameters\\": [{ \\"slug\\": \\"email\\", \\"type\\": \\"text\\", \\"required\\": true }],
      \\"tableId\\": \\"member\\",
      \\"vields\\": [
        {
          \\"id\\": \\"email\\",
          \\"name\\": \\"email\\",
          \\"type\\": \\"text\\",
          \\"deletionProtection\\": true,
          \\"createdAt\\": \\"2020-12-08T07:31:04.801Z\\"
        },
        {
          \\"id\\": \\"role\\",
          \\"name\\": \\"role\\",
          \\"type\\": \\"role\\",
          \\"deletionProtection\\": true,
          \\"createdAt\\": \\"2020-12-08T07:31:04.801Z\\"
        },
        {
          \\"id\\": \\"totalTask\\",
          \\"name\\": \\"Total task\\",
          \\"type\\": \\"rollup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:37:51.822Z\\"
        },
        {
          \\"id\\": \\"totalPoint\\",
          \\"name\\": \\"Total point\\",
          \\"type\\": \\"rollup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:38:17.873Z\\"
        },
        {
          \\"id\\": \\"totalHardTask\\",
          \\"name\\": \\"Total hard task\\",
          \\"type\\": \\"rollup\\",
          \\"deletionProtection\\": false,
          \\"createdAt\\": \\"2020-12-10T03:42:13.651Z\\"
        }
      ]
    }
  ]
}
"
`;
